src/js/Search/SearchVideoWrapper.js

```diff
@@ -0,0 +1,167 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable no-param-reassign */
+import { CLASSNAME, NUMBER, SNACKBAR_TEXT } from "../constants/index.js";
+import { messenger, MESSAGE } from "../messenger/index.js";
+import { renderSearchVideo, SEARCH_VIDEO_TEMPLATE } from "../Video/index.js";
+import { $, fetchYoutubeData, showModalSnackbar } from "../utils/index.js";
+
+export default class SearchVideoWrapper {
+  constructor() {
+    this.initializeVariables();
+    this.selectHTMLElements();
+    this.addMessageListeners();
+    this.addEventListeners();
+  }
+
+  initializeVariables() {
+    this.currentQuery = "";
+    this.currentNextPageToken = "";
+    this.videoItemsMap = new Map();
+  }
+
+  selectHTMLElements() {
+    this.$searchVideoWrapper = $(`.${CLASSNAME.SEARCH_VIDEO_WRAPPER}`);
+    this.$notFoundImg = $(`.${CLASSNAME.NOT_FOUND_IMAGE}`);
+  }
+
+  addMessageListeners() {
+    messenger.addMessageListener(MESSAGE.KEYWORD_SUBMITTED, ({ query }) => {
+      this.currentQuery = query;
+
+      $.hide(this.$notFoundImg);
+      $.clear(this.$searchVideoWrapper);
+
+      this.mountTemplate();
+      this.$searchVideoWrapper.scrollTo({ top: 0 });
+    });
```

Comment:
keyword 가 submit 되었을 때 하는 행동들이 한 눈에 안들어오는 것 같아요

추상화된 함수로 추출하여 가독성을 더 높여볼까요?

Reply:
한 줄 한 줄이 의미가 있는 코드들인데, 형태가 다 달라서 눈에 들어오지 않는 것 같습니다..

아래와 같이 수정하였습니다.

```suggestion
    messenger.addMessageListener(MESSAGE.KEYWORD_SUBMITTED, ({ query }) => {
      this.#currentQuery = query;

      this.#$searchVideoWrapper.innerHTML = "";

      this.#mountTemplate();

      this.#$searchVideoWrapper.scrollTo({ top: 0 });
    });
```